deckhouse-controller/pkg/controller/deckhouse-release/check_release.go

```diff
@@ -509,34 +526,31 @@ func (dcr *DeckhouseReleaseChecker) CalculateReleaseDelay(ts metav1.Time, cluste
 	return nil
 }
 
-func (dcr *DeckhouseReleaseChecker) StepByStepUpdate(actual, target *semver.Version) error {
+func (dcr *DeckhouseReleaseChecker) StepByStepUpdate(actual, target *semver.Version) (*semver.Version, error) {
 	nextVersion, err := dcr.nextVersion(actual, target)
 	if err != nil {
-		return err
-	}
-	if nextVersion == target {
-		return nil
+		return nil, fmt.Errorf("get next version: %w", err)
 	}
 
 	image, err := dcr.registryClient.Image(nextVersion.Original())
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("get image: %w", err)
 	}
 
 	releaseMeta, err := dcr.fetchReleaseMetadata(image)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("fetch release metadata: %w", err)
 	}
 	if releaseMeta.Version == "" {
-		return fmt.Errorf("version not found. Probably image is broken or layer does not exist")
+		return nil, fmt.Errorf("version not found. Probably image is broken or layer does not exist")
 	}
 
 	dcr.releaseMetadata = releaseMeta
 
-	return nil
+	return nextVersion, nil
 }
 
-func (dcr *DeckhouseReleaseChecker) nextVersion(actual, target *semver.Version) (*semver.Version, error) {
+func (dcr *DeckhouseReleaseChecker) nextVersion(actual, target *semver.Version) (next *semver.Version, err error) {
```

Comment:
Here we define two variables that will be at the output of the function: next, err
But I don't see their use in the function itself. Why do we define them then?